spec/apps/partial-rerouting/partial-rerouting.spec.js

```diff
@@ -0,0 +1,130 @@
+import * as singleSpa from "single-spa";
+
+describe("partial rerouting", () => {
+  function delay(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  beforeAll(() => singleSpa.start());
+
+  beforeEach(() => {
+    location.hash = "#";
+  });
+
+  it("shoulnd't bootstrap and mount app if the route has been changed while app was loading, but if the user navigates back it should be bootstrapped and mounted successfully", async () => {
+    const activeHash = "#partial-rerouting";
+
+    let bootstrapped = false;
+    let mounted = false;
+    let unmounted = false;
+
+    const app = {
+      async bootstrap() {
+        bootstrapped = true;
+      },
+      async mount() {
+        mounted = true;
+      },
+      async unmount() {
+        unmounted = true;
+      },
+    };
+
+    singleSpa.registerApplication(
+      "partial-rerouting",
+      () => delay(1000).then(() => app),
+      (location) => location.hash === activeHash
+    );
+
+    location.hash = activeHash;
+
+    const promise = singleSpa.triggerAppChange();
+    await delay(100);
+
+    singleSpa.navigateToUrl("#another-url");
+    await Promise.all([singleSpa.triggerAppChange(), promise]);
+
+    expect(bootstrapped).toBeFalsy();
+    expect(mounted).toBeFalsy();
+
```

Comment:
for clarity's sake: what is the expectation of what `unmounted` should be at this point?

should we add it here just to make sure it is what we expect it to be?

Reply:
Added expectation for `unmounted`.

Reply:
oh do you mind adding it as well to line 54? not a big deal, but I think once that's in then we're good.